pkg/loop/internal/test/test_plugin.go

```diff
@@ -0,0 +1,100 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/utils/tests"
+)
+
+type HelperProcessCommand struct {
+	Limit           int
+	CommandLocation string
+	Command         string
+}
+
+func WithLimit(limit int) func(*HelperProcessCommand) {
+	return func(h *HelperProcessCommand) {
+		h.Limit = limit
+	}
+}
+
+func (h HelperProcessCommand) New() *exec.Cmd {
+	cmdArgs := []string{
+		"go", "run", h.CommandLocation, fmt.Sprintf("-cmd=%s", h.Command),
+	}
+	if h.Limit != 0 {
+		cmdArgs = append(cmdArgs, fmt.Sprintf("-limit=%d", h.Limit))
+	}
+	cmd := exec.Command(cmdArgs[0], cmdArgs[1:]...) // #nosec
+	cmd.Env = os.Environ()
+	return cmd
+}
+
+func NewHelperProcess(commandLocation, command string, opts ...func(*HelperProcessCommand)) *exec.Cmd {
```

Comment:
Hmmm why functional options? I actually think these are quite problematic and not really necessary for an internal API.

Reply:
Why do you think they're problematic? I've always found them to be a useful feature for this kind of use case.

The square I'm trying to circle here is making the limit optional while allowing the `commandLocation` to be specified once per package somewhere.

Reply:
Hmmm... I'm not really sure how to address this comment in that case, since as I said this was the cleanest way I could think of to allow the limit to be optional while passing in a commandLocation.

Is this a blocking point for you? If so, can you give me a sense of how you would like to fix this?

Reply:
I'm not sure that I understand the problem. Why is the other API insufficient for doing this? Are you trying to prevent `commandLocation` from being set? I think that is still possible to validate in the helper func.

Reply:
Would something like this work in place of each helper func?
```go
type HelperProcessCommand struct {
	test.HelperProcessCommand
}

func (h HelperProcessCommand) New() *exec.Cmd {
	h.CommandLocation = "./internal/test/cmd/main.go"
	return h.HelperProcessCommand.New()
}
```

Reply:
I've refactored it so that it doesn't use functional options. Let me know what you think.

Reply:
LGTM :up:

Reply:
Er one small tweak to make the call sites cleaner.